wrapper/src/main/java/software/amazon/jdbc/plugin/cache/CacheConnection.java

```diff
@@ -61,7 +78,59 @@ public class CacheConnection {
           "true",
           "Whether to use SSL for cache connections.");
 
+  protected static final AwsWrapperProperty CACHE_IAM_AUTH_ENABLED =
+      new AwsWrapperProperty(
+          "cacheIamAuthEnabled",
+          "false",
+          "Whether to use IAM authentication for cache connections."
+      );
+
+  protected static final AwsWrapperProperty CACHE_IAM_REGION =
+      new AwsWrapperProperty(
+          "cacheIamRegion",
+          null,
+          "AWS region for ElastiCache IAM authentication.");
+
+  protected static final AwsWrapperProperty CACHE_IAM_USERNAME =
+      new AwsWrapperProperty(
+          "cacheIamUsername",
+          null,
+          "Username for ElastiCache IAM authentication.");
+
+  protected static final AwsWrapperProperty CACHE_HOST_NAME =
+      new AwsWrapperProperty(
+          "cacheHostName",
+          null,
+          "Cache hostname for IAM authentication.");
+
+  protected static final AwsWrapperProperty CACHE_PORT =
+      new AwsWrapperProperty(
+          "cachePort",
+          "6379",
+          "Cache port for IAM authentication.");
+
+  protected static final AwsWrapperProperty CACHE_USERNAME =
+      new AwsWrapperProperty(
+          "cacheUsername",
+          null,
+          "Username for ElastiCache regular authentication.");
+
+  protected static final AwsWrapperProperty CACHE_PASSWORD =
+      new AwsWrapperProperty(
+          "cachePassword",
+          null,
+          "Password for ElastiCache regular authentication.");
+
+  // Added because AWS generates some random string after your explicit cache name for uniqueness
+  protected static final AwsWrapperProperty CACHE_NAME =
```

Comment:
Not sure what this is. We should not need to make this into a custom property

Reply:
More specifically, you can parse it from the `cacheEndpointAddrRw` to start with?